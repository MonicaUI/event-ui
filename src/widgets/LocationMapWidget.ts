import { connect } from 'react-redux';
import LocationMapWidget from 'event-widgets/lib/LocationMap/LocationMapWidget';
import { getAddressWithCountryName } from 'event-widgets/redux/selectors/countryName';
import { getEventLocationMapType } from 'event-widgets/redux/selectors/account';
import { migrateMapTypeIsNeeded } from 'event-widgets/lib/LocationMap/LocationMapUtils';

export default connect((state: $TSFixMe) => {
  const {
    text: { translate }
  } = state;

  const addressWithCountryName = getAddressWithCountryName(state, translate);

  const locationMapType = getEventLocationMapType(state);

  return {
    location: translate(state.event.location),
    address: addressWithCountryName,
    googleApiKey: state.googleMap.apiKey,
    flexEventLocationMapType: migrateMapTypeIsNeeded(locationMapType),
    appleMapToken: state.appleMap.token
  };
})(LocationMapWidget);
