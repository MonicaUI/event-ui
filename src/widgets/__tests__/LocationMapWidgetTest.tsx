import React from 'react';
import LocationMapWidget from '../LocationMapWidget';
import { shallow } from 'enzyme';

const subscribe = jest.fn();

function getState() {
  return {
    countries: {
      countries: {
        US: {
          code: 'US',
          sort: 1,
          name: 'USA',
          nameResourceKey: 'cvt_lu2_0229',
          isoCode: 840,
          requireZipCodeFlag: true,
          metroAreaGroupId: 0,
          id: 232,
          alphaThreeCode: 'USA'
        }
      }
    },
    text: {
      translate: text => text
    },
    event: {
      location: 'New York',
      address: {
        address1: '2900 Private Drive',
        city: 'Atlanta',
        stateCode: 'GA',
        postalCode: '22203',
        countryCode: 'US'
      }
    },
    googleMap: {
      apiKey: 'DummyKeyString'
    },
    appleMap: {
      token: 'DummyAppleMapToken'
    },
    account: {
      settings: {
        flexEventLocationMapType: 2
      }
    }
  };
}

async function dispatch(action) {
  if (typeof action === 'function') {
    await action(dispatch, getState);
  }
}

const googleMapDefaultProps = {
  store: { dispatch, getState, subscribe },
  text: {
    translate: text => text
  },
  event: {
    location: 'New York',
    address: {
      address1: '2900 Private Drive',
      city: 'Atlanta',
      stateCode: 'GA',
      postalCode: '22203',
      countryCode: 'US'
    }
  },
  googleMap: {
    apiKey: 'DummyKeyString'
  },
  appleMap: {
    token: 'DummyAppleMapToken'
  },
  account: {
    settings: {
      flexEventLocationMapType: 1
    }
  }
};
const appleMapDefaultProps = {
  store: { dispatch, getState, subscribe },
  guestText: {
    translate: text => text
  },
  event: {
    location: 'New York',
    address: {
      address1: '2900 Private Drive',
      city: 'Atlanta',
      stateCode: 'GA',
      postalCode: '22203',
      countryCode: 'US'
    }
  },
  googleMap: {
    apiKey: 'DummyKeyString'
  },
  appleMap: {
    token: 'DummyAppleMapToken'
  },
  account: {
    settings: {
      flexEventLocationMapType: 1
    }
  }
};
describe('LocationMapWidget', () => {
  it('should render Google Map with address containing country name', () => {
    // @ts-expect-error ts-migrate(2559) FIXME: Type '{ store: { dispatch: (action: any) => Promis... Remove this comment to see the full error message
    const widget = shallow(<LocationMapWidget {...googleMapDefaultProps} />);
    expect(widget.props()).toMatchSnapshot();
  });

  it('should render Apple Map with address containing country name', () => {
    // @ts-expect-error ts-migrate(2559) FIXME: Type '{ store: { dispatch: (action: any) => Promis... Remove this comment to see the full error message
    const widget = shallow(<LocationMapWidget {...appleMapDefaultProps} />);
    expect(widget.props()).toMatchSnapshot();
  });
});
