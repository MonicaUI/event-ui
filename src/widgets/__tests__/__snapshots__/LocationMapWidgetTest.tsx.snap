// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationMapWidget should render Apple Map with address containing country name 1`] = `
Object {
  "children": <LocationMapWidget
    account={
      Object {
        "settings": Object {
          "flexEventLocationMapType": 1,
        },
      }
    }
    address={
      Object {
        "address1": "2900 Private Drive",
        "city": "Atlanta",
        "countryCode": "US",
        "countryName": "cvt_lu2_0229",
        "postalCode": "22203",
        "stateCode": "GA",
      }
    }
    appleMap={
      Object {
        "token": "DummyAppleMapToken",
      }
    }
    appleMapToken="DummyAppleMapToken"
    dispatch={[Function]}
    event={
      Object {
        "address": Object {
          "address1": "2900 Private Drive",
          "city": "Atlanta",
          "countryCode": "US",
          "postalCode": "22203",
          "stateCode": "GA",
        },
        "location": "New York",
      }
    }
    flexEventLocationMapType={2}
    googleApiKey="DummyKeyString"
    googleMap={
      Object {
        "apiKey": "DummyKeyString",
      }
    }
    guestText={
      Object {
        "translate": [Function],
      }
    }
    location="New York"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [MockFunction],
      }
    }
  />,
  "value": null,
}
`;

exports[`LocationMapWidget should render Google Map with address containing country name 1`] = `
Object {
  "children": <LocationMapWidget
    account={
      Object {
        "settings": Object {
          "flexEventLocationMapType": 1,
        },
      }
    }
    address={
      Object {
        "address1": "2900 Private Drive",
        "city": "Atlanta",
        "countryCode": "US",
        "countryName": "cvt_lu2_0229",
        "postalCode": "22203",
        "stateCode": "GA",
      }
    }
    appleMap={
      Object {
        "token": "DummyAppleMapToken",
      }
    }
    appleMapToken="DummyAppleMapToken"
    dispatch={[Function]}
    event={
      Object {
        "address": Object {
          "address1": "2900 Private Drive",
          "city": "Atlanta",
          "countryCode": "US",
          "postalCode": "22203",
          "stateCode": "GA",
        },
        "location": "New York",
      }
    }
    flexEventLocationMapType={2}
    googleApiKey="DummyKeyString"
    googleMap={
      Object {
        "apiKey": "DummyKeyString",
      }
    }
    location="New York"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [MockFunction],
      }
    }
    text={
      Object {
        "translate": [Function],
      }
    }
  />,
  "value": null,
}
`;
